### ISSUE TITLE
Mutation with (remote [env] false) seems to cause some kind of infinite tx loop

### ISSUE THREAD

**Issue by nivekuil** [fulcrologic/fulcro#433]
Tested with this, 3.4.3:
```
(m/defmutation test [_]
     (action [env] (log/info "test"))
     (remote [env] false))
(comp/transact! app.application/SPA `[(test)])
```
When transact runs, it blasts the console with output like this:
```
(... and so on)
core.cljc:179 2020-10-27T03:48:20.310Z INFO [app.model.view:2] - test
core.cljc:179 2020-10-27T03:48:20.312Z INFO [app.model.view:2] - test
core.cljc:179 2020-10-27T03:48:20.313Z INFO [app.model.view:2] - test
core.cljc:179 2020-10-27T03:48:20.315Z INFO [app.model.view:2] - test
core.cljc:179 2020-10-27T03:48:20.317Z INFO [app.model.view:2] - test
core.cljc:179 2020-10-27T03:48:20.319Z INFO [app.model.view:2] - test
core.cljc:179 2020-10-27T03:48:20.321Z INFO [app.model.view:2] - test
react_devtools_backend.js:2450 2020-10-27T03:48:20.323Z ERROR [com.fulcrologic.fulcro.algorithms.tx-processing.synchronous-tx-processing:359] - Error processing tx queue!
RangeError: Maximum call stack size exceeded
    at eval (http://localhost:1234/js/main/cljs-runtime/taoensso.timbre.js:71:171)
    at Function.eval [as cljs$core$IFn$_invoke$arity$2] (http://localhost:1234/js/main/cljs-runtime/taoensso.timbre.js:75:3)
    at Function.eval [as cljs$core$IFn$_invoke$arity$1] (http://localhost:1234/js/main/cljs-runtime/taoensso.timbre.js:33:42)
    at Function.eval [as cljs$core$IFn$_invoke$arity$3] (http://localhost:1234/js/main/cljs-runtime/cljs.core.js:13259:10)
    at Function.eval [as cljs$core$IFn$_invoke$arity$2] (http://localhost:1234/js/main/cljs-runtime/cljs.core.js:13252:34)
    at Function.eval [as cljs$core$IFn$_invoke$arity$2] (http://localhost:1234/js/main/cljs-runtime/cljs.core.js:13534:34)
    at G__265638__delegate (http://localhost:1234/js/main/cljs-runtime/cljs.core.js:35000:27)
    at G__265638 (http://localhost:1234/js/main/cljs-runtime/cljs.core.js:35015:28)
    at eval (http://localhost:1234/js/main/cljs-runtime/taoensso.timbre.js:768:114)
    at Object.eval [as cljs$core$IDeref$_deref$arity$1] (http://localhost:1234/js/main/cljs-runtime/cljs.core.js:34123:111)
react_devtools_backend.js:2450 2020-10-27T03:48:20.344Z ERROR [com.fulcrologic.fulcro.algorithms.tx-processing.synchronous-tx-processing:326] - Old queue changed!
react_devtools_backend.js:2450 2020-10-27T03:48:20.344Z ERROR [com.fulcrologic.fulcro.algorithms.tx-processing.synchronous-tx-processing:326] - Old queue changed!
react_devtools_backend.js:2450 2020-10-27T03:48:20.344Z ERROR [com.fulcrologic.fulcro.algorithms.tx-processing.synchronous-tx-processing:326] - Old queue changed!
react_devtools_backend.js:2450 2020-10-27T03:48:20.344Z ERROR [com.fulcrologic.fulcro.algorithms.tx-processing.synchronous-tx-processing:326] - Old queue changed!
(... and so on)
```
Removing the (remote) section makes it behave as expected. For reference, the book claims that returning false should be the same as not having the (remote) section at all.

**Comment by nivekuil** [comment]
Seems to happen only with `stx/with-synchronous-transactions`

**Comment by awkay** [comment]
I don't have time to work on this myself. Feel free to submit a PR if you find the cause.

**Comment by awkay** [comment]
Fixed in 3.2.14
